src/components/SignInPage.vue

```diff
@@ -10,20 +11,31 @@ import { ref } from 'vue';
 
 // Create a reactive reference for the username
 const username = ref('');
+const password = ref('');
 
 // Handle the login action
 async function handleLogin () {
   // Check if the username input is not empty after trimming whitespace
   if (username.value.trim()) {
     try {
       // Construct the URL with the username
-      const url = `http://localhost:8080/${username.value.trim()}`;
+      const url = `http://localhost:8080/login`;
 
-      let response = await fetch(url)
+      let data = {
+        "password":  password.value.trim(),
+        "username":  username.value.trim()
+      }
 
-      console.log(response)
+      let response = await fetch(url, {
+        headers: {
+                'Content-Type': 'application/json'
+            },
+        method: 'POST',
+        body: JSON.stringify(data)})
+
+    let responseData = await response.json()
```

Comment:
next step is checking the response from the api 

https://www.tutorialspoint.com/ajax/fetch_api_status_codes.htm


if the response is ok / status code 200 then take the user to the home page, if it's not update the text on the login page to indicate that their username or password is incorrect